modules/backend/controllers/Files.php

```diff
@@ -64,33 +65,33 @@ public function thumb($code = null, $width = 100, $height = 100, $mode = 'auto',
      */
     protected static function getTemporaryUrl($file, $path = null)
     {
-        // Get the disk and adapter used
-        $url = null;
+        // Get the disk used
         $disk = $file->getDisk();
-        $adapter = $disk->getAdapter();
-        if (class_exists('\League\Flysystem\Cached\CachedAdapter') && $adapter instanceof \League\Flysystem\Cached\CachedAdapter) {
-            $adapter = $adapter->getAdapter();
-        }
 
-        if ((class_exists('\League\Flysystem\AwsS3v3\AwsS3Adapter') && $adapter instanceof \League\Flysystem\AwsS3v3\AwsS3Adapter) ||
-            (class_exists('\League\Flysystem\Rackspace\RackspaceAdapter') && $adapter instanceof \League\Flysystem\Rackspace\RackspaceAdapter) ||
-            method_exists($adapter, 'getTemporaryUrl')
-        ) {
-            if (empty($path)) {
-                $path = $file->getDiskPath();
-            }
+        if (empty($path)) {
+            $path = $file->getDiskPath();
+        }
 
-            // Check to see if the URL has already been generated
-            $pathKey = 'backend.file:' . $path;
-            $url = Cache::get($pathKey);
+        // Check to see if the URL has already been generated
+        $pathKey = 'backend.file:' . $path;
+        $url = Cache::get($pathKey, null);
 
-            // The AWS S3 storage drivers will return a valid temporary URL even if the file does not exist
-            if (is_null($url) && $disk->exists($path)) {
-                $expires = now()->addSeconds(Config::get('cms.storage.uploads.temporaryUrlTTL', 3600));
-                $url = Cache::remember($pathKey, $expires, function () use ($disk, $path, $expires) {
+        if ($url !== false && is_null($url) && $disk->exists($path)) {
```

Comment:
This condition looks a bit wonky. There's no need for the `$url !== false` if we're checking that `$url` is `null`.

```suggestion
        if (is_null($url) && $disk->exists($path)) {
```